rllib/agents/slateq/slateq_torch_model.py

```diff
@@ -13,71 +13,64 @@
 
 
 class QValueModel(nn.Module):
-    """The Q-value model for SlateQ.
-
-    A simple MLP with n layers (w/ LeakyReLU activation), whose sizes are
-    specified via `config.q_hiddens`.
-    """
-
-    def __init__(self, embedding_size: int, q_hiddens: Sequence[int]):
+    def __init__(
+        self,
+        obs_space: gym.spaces.Space,
+        fcnet_hiddens_per_candidate=(256, 32),
+    ):
         """Initializes a QValueModel instance.
 
-        Args:
-            embedding_size: The sum of user- and doc embedding sizes. This is the input
-                dimension going into the first layer of the MLP.
-            q_hiddens: List of dense layer sizes to build the MLP by.
+        Each document candidate receives one full Q-value stack, defined by
+        `fcnet_hiddens_per_candidate`. The input to each of these Q-value stacks
+        is always {[user] concat [document[i]] for i in document_candidates}.
+
+        Extra model kwargs:
+            fcnet_hiddens_per_candidate: List of layer-sizes for each(!) of the
+                candidate documents.
         """
         super().__init__()
 
-        # Construct hidden layers.
-        layers = []
-        ins = embedding_size
-        for n in q_hiddens:
-            layers.append(nn.Linear(ins, n))
-            layers.append(nn.LeakyReLU())
-            ins = n
-        layers.append(nn.Linear(ins, 1))
-        self.layers = nn.Sequential(*layers)
+        self.orig_obs_space = obs_space
+        self.embedding_size = self.orig_obs_space["doc"]["0"].shape[0]
+        self.num_candidates = len(self.orig_obs_space["doc"])
+        assert self.orig_obs_space["user"].shape[0] == self.embedding_size
```

Comment:
should we add a message here to go with the assert? Or is this more a sanity check that the user probably won't hit.